cf/data/data.py

```diff
@@ -5914,9 +6179,13 @@ def size(self):
 
         """
         dx = self._get_dask()
-        return dx.size
+        size = dx.size
+        if math.isnan(size):
```

Comment:
I don't know! dask itself uses both, in various places, and I'm not sure what the difference is, `math.isnan` is ~4 times faster when size  _is_ nan, but they're about the same speed when size is an integer . So I guess `math.isnan` is a good choice in general in such circumstances ...

Reply:
Fair enough, I guess the question arose because I would go for `np.isnan` by default.

> So I guess math.isnan is a good choice in general in such circumstances ...

Shall we try to do this for now (`math.isnan` if it covers all bases for a given case, else use `np`) and I will make a note that towards the end of the daskification, as part of tidying work, we can quickly review which is used in any `.isnan` case we end up needing (as you point out it may not be significant but it would also take ~5 mins to change, so )?